File: match-service/internal/models/player.go
package models

type Player struct {
	ID       string `json:"id"`
	Name     string `json:"name"`
	Role     string `json:"Role"`
	Position int    `json:"position"`
	TicketID string `json:"ticketID"`
}

type PlayerInfo struct {
	ID   string `json:"id"`
	Name string `json:"name"`
	Role string `json:"role"`
}

File: matchmaking-service/Models/Player.cs
using System.Text.Json.Serialization;

public record Player(string Id, string Name, string Role, string TicketID)
{
    [JsonIgnore]
    public bool IsKiller => Role == "killer";
    [JsonIgnore]
    public bool IsSurvivor => Role == "survivor";
    public PlayerInfo ToPlayerInfo() => new(Id, Name, Role);
};

public record PlayerInfo(string Id, string Name, string Role)
{
    public Player ToPlayer() => new(Id, Name, Role, Guid.NewGuid().ToString());
};
